### PR TITLE
feat: refactor notebook and supporting code

### PR DESCRIPTION
**Pull request by dwinston**
# Description

This is a refactored referential-integrity-checker prototype notebook based on requested documentation. In many cases where justification for a decision was requested, the justification was deemed to be historical and no longer applicable, prompting a refactor.

Careful review is requested.

Fixes # 604

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] run-through of notebook

# Definition of Done (DoD) Checklist:

- [x] I have performed a self-review of my code
- [x] New and existing unit and functional tests pass locally with my changes (`make up-test && make test-run`)

### CODE REVIEW

**Comment by PeopleMakeCulture** [comment]
> @dwinston did you see the error for the failing CI test? Looks like something's wrong with OAuth -- perhaps Orchid restructured the response JSON?
> 
> ```
> tests/test_api/test_endpoints.py:150: 
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> nmdc_runtime/site/resources.py:60: in request
>     self.ensure_token()
> nmdc_runtime/site/resources.py:43: in ensure_token
>     self.get_token()
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> 
> self = <nmdc_runtime.site.resources.RuntimeApiSiteClient object at 0x7f6a4ebeb010>
> 
>     def get_token(self):
>         rv = requests.post(self.base_url + "/token", data=self.get_token_request_body())
>         self.token_response = rv.json()
>         if "access_token" not in self.token_response:
> >           raise Exception(f"Getting token failed: {self.token_response}")
> E           Exception: Getting token failed: {'detail': [{'type': 'missing', 'loc': ['query', 'extra_kwargs'], 'msg': 'Field required', 'input': None}]}
> ```

Saw the update to kwargs in the 2nd commit. I will re-run the tests

**Comment by PeopleMakeCulture** [comment]
> > @dwinston did you see the error for the failing CI test? Looks like something's wrong with OAuth -- perhaps Orchid restructured the response JSON?
> > ```
> > tests/test_api/test_endpoints.py:150: 
> > _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> > nmdc_runtime/site/resources.py:60: in request
> >     self.ensure_token()
> > nmdc_runtime/site/resources.py:43: in ensure_token
> >     self.get_token()
> > _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> > 
> > self = <nmdc_runtime.site.resources.RuntimeApiSiteClient object at 0x7f6a4ebeb010>
> > 
> >     def get_token(self):
> >         rv = requests.post(self.base_url + "/token", data=self.get_token_request_body())
> >         self.token_response = rv.json()
> >         if "access_token" not in self.token_response:
> > >           raise Exception(f"Getting token failed: {self.token_response}")
> > E           Exception: Getting token failed: {'detail': [{'type': 'missing', 'loc': ['query', 'extra_kwargs'], 'msg': 'Field required', 'input': None}]}
> > ```
> 
> Saw the update to kwargs in the 2nd commit. I will re-run the tests

Tests are still failing

File: docker-compose.test.yml
Comment (PeopleMakeCulture) [inline docker-compose.test.yml:1]: Are you changing the test mongodb to point to a local dump from mongorestore?

File: nmdc_runtime/api/db/mongo.py [hunk 1]
```diff
@@ -74,3 +95,14 @@ def activity_collection_names(mdb: MongoDatabase) -> Set[str]:
         "functional_annotation_set",
         "genome_feature_set",
     }
+
+
```
Comment (PeopleMakeCulture) [inline nmdc_runtime/api/db/mongo.py:99]: Add documentation for why this is required

**Review by PeopleMakeCulture — APPROVED** [review]

File: docker-compose.test.yml
Comment (dwinston) [reply docker-compose.test.yml:1]: This change was to ensure that, even on a localhost dev machine, the mongo and dagster db used for testing are isolated (by having different named docker volumes) from those used for dev.

File: nmdc_runtime/api/db/mongo.py (hunk 1 above)
Comment (dwinston) [reply nmdc_runtime/api/db/mongo.py:99]: I assume this question is re: the addition of the `mongodump_excluded_collections()` helper. This was used to sub-select from a full database dump to ensure only needed collections for testing referential integrity code and avoiding storing personally identifiable information (PII) as obtained from the full dumps.

However, this helper more properly represents unfinished work to enable regular posting of "public" backups of nmdc metadata to streamline CI, dev, etc. If I had to do it over again, I'd have left this code out of this PR and reserved it for a separate dedicated feature branch.

**Pull request merged**